commit 7c9f19e9f547a6ef9c54be808aea8663f91876bc
Author: tnut <tnut@nutyx.org>
Date:   Fri Mar 22 15:04:41 2019 +0100

    cards upgrade should install missing groups/locales packages when missing

diff --git a/src/cards_upgrade.cxx b/src/cards_upgrade.cxx
index 7d9facf..9bd1a44 100644
--- a/src/cards_upgrade.cxx
+++ b/src/cards_upgrade.cxx
@@ -36,6 +36,29 @@ Cards_upgrade::Cards_upgrade(const CardsArgumentParser& argParser,
 
 	parsePkgRepoCollectionFile();
 	buildSimpleDatabase();
+	std::set<std::string> tmpList;
+	if (!m_config.group.empty()) {
+		for (auto i: m_listOfInstPackages) {
+			for ( auto j :  m_config.group ) {
+					std::string packageName  = getBasePackageName(i.first) + "." + j;
+					if ( i.first == packageName )
+						continue;
+					if (checkBinaryExist(packageName)) {
+						tmpList.insert(packageName);
+					}
+			}
+		}
+	}
+	if ( tmpList.size() > 0) {
+		for ( auto i : tmpList) {
+			pair<string,time_t> packageNameBuildDate;
+			packageNameBuildDate.first = i;
+			packageNameBuildDate.second = getBinaryBuildTime(i);
+			if (checkPackageNameBuildDateSame(packageNameBuildDate))
+				continue;
+			m_ListOfPackages.insert(packageNameBuildDate);
+		}
+	}
 	for (auto i : m_listOfInstPackages) {
 		if (!checkBinaryExist(i.first)) {
 			cerr << i.first << " not exist" << endl;
